File: 2022-2023/sourcecode/lab/lab4/JetpackNavigationJava/app/src/main/java/pl/edu/uwr/pum/jetpacknavigationjava/FragmentA.java
package pl.edu.uwr.pum.jetpacknavigationjava;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

public class FragmentA extends Fragment {
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_a, container, false);

        view.findViewById(R.id.fabA).setOnClickListener(v ->{
            //NavDirections action = FragmentADirections.toFragmentB();

            Bundle args = new Bundle();
            args.putInt("key", 5);
            Navigation.findNavController(view).navigate(R.id.to_fragmentB, args);
        });

        return view;
    }
}

File: 2022-2023/sourcecode/lab/lab4/JetpackNavigationKotlin/app/src/main/java/pl/edu/uwr/pum/jetpacknavigationkotlin/FragmentA.kt
package pl.edu.uwr.pum.jetpacknavigationkotlin

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.Navigation
import com.google.android.material.floatingactionbutton.FloatingActionButton

class FragmentA : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_a, container, false)

        view.findViewById<FloatingActionButton>(R.id.fabA).setOnClickListener {
            //val action = FragmentADirections.toFragmentB()

            val args = Bundle()
            args.putInt("key", 5)
            Navigation.findNavController(view).navigate(R.id.to_fragmentB, args)
        }

        return view
    }
}